### PR TITLE
Add tests and python code

### PR DESCRIPTION
**Pull request by olgabot**
Added a bunch of files, largely from https://github.com/audreyr/cookiecutter-pypackage

Some questions/comments:
- Should we set up docs auto-deployment to the codeneuro.github.io/gitgoing page?
- PyPI (Python Package Index) requires a README.rst, not README.md file and I haven't found success with using markdown --> RST converters, so I suggest creating a barebones RST file for PyPI deployment

### CODE REVIEW

File: gitgoing/gitgoing.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+NEURON_COMPONENTS = ('soma', 'axon', 'dendrite', 'synapse')
+
+def mean(x):
```
Comment (freeman-lab) [inline gitgoing/gitgoing.py]: Maybe avoid overloading these function names from numpys? Something cheesy like `my_mean` and `my_std` might work well for this purpose.

Comment (freeman-lab) [inline gitgoing/gitgoing.py]: Or even just call it `mean_plus_one`, might actually help clarify the whole plus 1 business

**Comment by freeman-lab** [comment]
This is all looking awesome @olgabot ! A few things:
- I enabled tests on Travis, lemme know if it's not working
- Tests are great! I had one suggestion about naming
- Is it over-complicating to assure functional tests on Python 2.x and 3.x? Are we expecting people to test locally with both versions? My inclination would be to do everything in 2.7+ for this tutorial, and maybe explain how to handle other versions near the end.
- I've hit that thing with README.rst as well in my own projects, I think a dummy README.rst that basically just links to the github repo or project page is perfectly fine
- For the sphinx page, it might be worth letting people see how it works manually. I actually just set up the auto deployment you suggested over on Thunder (thanks again for the tip!), and while I love it, it does a lot of git kung foo behind the scenes, and obscures the sphinx building process. I'm glad I learned how to do it manually, for the purpose of future debugging, so maybe that's one to walk people through. Seem reasonable?

Maybe after addressing these couple things (and fixing whatever the merge conflict is) we can merge this in, on board @bensussman ?

**Comment by bensussman** [comment]
I'm on board with this PR! I'll have to take it home (maybe tomorrow afternoon as it's the only time i'm really free this week) and actually exercise it on my mac to see if it all fits together. :+1:

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+#texinfo_appendices = []
+
+# If false, no module index is generated.
+#texinfo_domain_indices = True
+
+# How to display URL addresses: 'footnote', 'no', or 'inline'.
+#texinfo_show_urls = 'footnote'
+
+# If true, do not generate a @detailmenu in the "Top" node's menu.
+#texinfo_no_detailmenu = False
```
Comment (bensussman) [inline docs/conf.py:275]: isn't this file generated? should it be included in git?

File: docs/make.bat [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+if "%1" == "pseudoxml" (
+	%SPHINXBUILD% -b pseudoxml %ALLSPHINXOPTS% %BUILDDIR%/pseudoxml
+	if errorlevel 1 exit /b 1
+	echo.
+	echo.Build finished. The pseudo-XML files are in %BUILDDIR%/pseudoxml.
+	goto end
+)
+
+:end
```
Comment (bensussman) [inline docs/make.bat:242]: this file is huge and scary! What are our expectations in terms of the students:
1.  understanding these files, or knowing to not open them and `cp` them into their future projects
2.  understanding what sphinx docs do and how they work, why, and when.

I think (at least for me personally) this is a meaty topic worthy of some explanation slides, some lecture component, and some interact-able examples.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        'Development Status :: 2 - Pre-Alpha',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: BSD License',
+        'Natural Language :: English',
+        "Programming Language :: Python :: 2",
+        'Programming Language :: Python :: 2.6',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
```
Comment (bensussman) [inline setup.py:56]: non `2.7` python versions show up in a few places in this PR. Is that intentional? Are we intending this code to support lots of versions? Is that something we want to do? is it even possible to have python code that is cross compatible across the `2->3` boundary?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Config file for automatic testing at travis-ci.org
+
+language: python
+
+python:
+  - "3.4"
```
Comment (bensussman) [inline .travis.yml]: more python version craziness...

**Comment by freeman-lab** [comment]
LGTM!

**Pull request merged**